src/Illuminate/Database/Query/Builder.php

```diff
@@ -3949,9 +3949,9 @@ public function upsert(array $values, $uniqueBy, $update = null)
 
         $bindings = $this->cleanBindings(array_merge(
             Arr::flatten($values, 1),
-            (new Collection($update))->reject(function ($value, $key) {
-                return is_int($key);
-            })->all()
+            (new Collection($update))
+                ->reject(fn ($value, $key) => is_int($key))
```

Comment:
This can be simplified to
```suggestion
                ->reject('is_int')
```

or
```suggestion
                ->reject(is_int(...))
```

Reply:
I know this is closed, but FYI this would have:

1. thrown an exception, as `is_int` would have received two arguments. Something we need to be mindful of when migrating to first-class callables
2. Even if this was a user-land function that did not throw on additional arguments, the  behaviour would have changed as now `$value` would be passed through instead of `$key`